apps/aecore/test/aecore_suite_utils.erl

```diff
@@ -343,20 +344,22 @@ mine_blocks_until_txs_on_chain_loop(_Node, _TxHashes, 0, _Acc) ->
 mine_blocks_until_txs_on_chain_loop(Node, TxHashes, Max, Acc) ->
     case mine_blocks_loop(1, key) of
         {ok, [Block]} -> %% We are only observing key blocks
+            ct:log("mined a key block", []),
             NewAcc = [Block | Acc],
-            case txs_not_on_chain(Node, Block, TxHashes) of
+            TxsNotOnChain = TxHashes -- txs_on_chain(Node, TxHashes),
+            case TxsNotOnChain of
                 []        -> {ok, NewAcc};
                 TxHashes1 -> mine_blocks_until_txs_on_chain_loop(Node, TxHashes1, Max - 1, NewAcc)
             end;
         {error, _} = Error -> Error
     end.
 
-txs_not_on_chain(Node, Block, TxHashes) ->
```

Comment:
This is quite a drastic change... I don't have the energy to figure out what is the problem with the existing approach? 

It will mean that we have to change the input to `mine_txs` everywhere?